chain/client/tests/chunks_management.rs

```diff
@@ -250,3 +252,137 @@ fn test_request_chunk_restart() {
         assert!(false);
     }
 }
+
+#[test]
+fn store_partial_encoded_chunk_sanity() {
+    init_test_logger();
+    let mut env = TestEnv::new(ChainGenesis::test(), 1, 1);
+    let signer = InMemorySigner::from_seed("test0", KeyType::ED25519, "test0");
+    let mut partial_encoded_chunk = PartialEncodedChunk {
+        shard_id: 0,
+        chunk_hash: Default::default(),
+        header: Some(ShardChunkHeader::new(
+            CryptoHash::default(),
+            CryptoHash::default(),
+            CryptoHash::default(),
+            CryptoHash::default(),
+            1,
+            1,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            CryptoHash::default(),
+            CryptoHash::default(),
+            vec![],
+            &signer,
+        )),
+        parts: vec![],
+        receipts: vec![],
+    };
+    let block_hash = env.clients[0].chain.genesis().hash();
+    let block = env.clients[0].chain.get_block(&block_hash).unwrap().clone();
+    assert_eq!(env.clients[0].shards_mgr.get_stored_partial_encoded_chunks(1).len(), 0);
+    env.clients[0]
+        .shards_mgr
+        .store_partial_encoded_chunk(&block.header, partial_encoded_chunk.clone());
+    assert_eq!(env.clients[0].shards_mgr.get_stored_partial_encoded_chunks(1).len(), 1);
+    assert_eq!(
+        env.clients[0].shards_mgr.get_stored_partial_encoded_chunks(1)[&0],
+        partial_encoded_chunk
+    );
+
+    // Check replacing
+    partial_encoded_chunk.chunk_hash.0 = hash(vec![123].as_ref());
+    env.clients[0]
+        .shards_mgr
+        .store_partial_encoded_chunk(&block.header, partial_encoded_chunk.clone());
+    assert_eq!(env.clients[0].shards_mgr.get_stored_partial_encoded_chunks(1).len(), 1);
+    assert_eq!(
+        env.clients[0].shards_mgr.get_stored_partial_encoded_chunks(1)[&0],
```

Comment:
How does this work, you changed the hash but not the signature, right?
Or is it becaues signature verification is disabled in tests?

Reply:
`verify_validator_signature` is always true in KV-runtime.